operators/pkg/controller/remotecluster/watches.go

```diff
@@ -47,15 +48,69 @@ func addWatches(c controller.Controller, r *ReconcileRemoteCluster) error {
 		return err
 	}
 
+	// Watch licenses in order to enable functionality if license status changes
+	if err := c.Watch(&source.Kind{Type: &v1alpha1.EnterpriseLicense{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: reconcileAllRemoteClusters(r.Client),
+	}); err != nil {
+		return err
+	}
+
+	// Watch Service resources in order to reconcile related remote clusters if it changes.
+	if err := c.Watch(&source.Kind{Type: &v1.Service{}}, &handler.EnqueueRequestsFromMapFunc{
+		// if it's a remote cluster seed service, we need to enqueue requests for any relevant remote clusters
+		ToRequests: handler.ToRequestsFunc(func(object handler.MapObject) []reconcile.Request {
+			if !strings.HasSuffix(object.Meta.GetName(), remoteClusterSeedServiceSuffix) {
+				// not a remote cluster seed service, safe to ignore
+				return nil
+			}
+
+			svcNamespace := object.Meta.GetNamespace()
+			svcName := object.Meta.GetName()
+
+			var list v1alpha1.RemoteClusterList
+			if err := r.List(&client.ListOptions{}, &list); err != nil {
+				log.Error(err, "failed to list remote clusters in watch handler")
+				// dropping any errors on the floor here
+				return nil
+			}
+
+			var reqs []reconcile.Request
+			for _, rc := range list.Items {
+				// compare service name
+				if remoteClusterSeedServiceName(rc.Spec.Remote.K8sLocalRef.Name) != svcName {
+					continue
+				}
+
+				// compare service namespace
+				ns := rc.Spec.Remote.K8sLocalRef.Namespace
+				if ns == "" {
+					ns = rc.Namespace
+				}
+				if ns != svcNamespace {
+					// Service and remote namespace in different namespaces, so not relevant
+					continue
+				}
+
+				log.Info("Synthesizing reconcile for ", "resource", k8s.ExtractNamespacedName(&rc))
+				reqs = append(reqs, reconcile.Request{
+					NamespacedName: k8s.ExtractNamespacedName(&rc),
+				})
+			}
+			return reqs
```

Comment:
Normally a test would be nice for code of this complexity, but given that we are about to retire the whole remote cluster thing, it is probably not worth the effort.

Reply:
Added tests for this and the finalizer.